apps/api/src/app/widgets/usecases/mark-all-messages-as/mark-all-messages-as.usecase.ts

```diff
@@ -52,18 +52,24 @@ export class MarkAllMessagesAs {
       channel: ChannelTypeEnum.IN_APP,
     });
 
-    const isMarkAsRead = command.markAs === 'read';
+    const isUnreadCountChanged =
+      command.markAs === MarkMessagesAsEnum.READ || command.markAs === MarkMessagesAsEnum.UNREAD;
+
+    const countQuery = isUnreadCountChanged ? { read: false } : { seen: false };
+
+    const count = await this.messageRepository.getCount(
+      command.environmentId,
+      subscriber._id,
+      ChannelTypeEnum.IN_APP,
+      countQuery
+    );
```

Comment:
Get a count of the "unread" or "unseen" messages. This value is passed with the event through the WebSocket.